### DISCUSSION TITLE
`vite-plugin-dts` vs `vue-tsc --declaration --emitDeclarationOnly`

### DISCUSSION THREAD

**Discussion by mesqueeb** [vuejs/language-tools#1740] [General]
I'm wondering about the pros and cons of each.

Personally, often `vue-tsc --declaration --emitDeclarationOnly` just works better.

The pro about `vite-plugin-dts` is that it has well documented configuration you can pass in the `vite.config.ts`.

Any other findings?

**Comment by johnsoncodehk** [comment]
Good question, I don't know :), maybe you can reference https://github.com/qmhc/vite-plugin-dts/search?q=%22vue-tsc%22&type=issues.

**Comment by catnuko** [comment]
vite-plugin-dts struggles to compile complex nested vue components
